cmd/controller-manager/app/controller_manager.go

```diff
@@ -111,21 +112,30 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		rest.AddUserAgent(k8sKubeconfig, controllerManagerAgentName),
 	)
 	if err != nil {
-		glog.Fatalf("Invalid k8s API configuration: %v", err)
+		return fmt.Errorf("invalid Kubernetes API configuration: %v", err)
```

Comment:
@pwittrock Thanks for digging through this. You are correct, somehow I mistakenly thought `glog.Fatalf` doesn't exit. I also agree returning is better vs having many exit points through out the codebase.